src/Features/CSharp/Portable/CodeRefactorings/EnableNullable/EnableNullableCodeRefactoringProvider.cs

```diff
@@ -62,6 +61,27 @@ public override async Task ComputeRefactoringsAsync(CodeRefactoringContext conte
                 new CustomCodeAction((purpose, cancellationToken) => EnableNullableReferenceTypesAsync(document.Project, purpose, context.Options, cancellationToken)));
         }
 
+        private static bool CanEnableNullableInProject(Project project)
+            => project is
+            {
+                ParseOptions: CSharpParseOptions { LanguageVersion: >= LanguageVersion.CSharp8 },
+                CompilationOptions.NullableContextOptions: NullableContextOptions.Disable,
+            };
+
+        private static async Task<Solution> EnableNullableReferenceTypesAsync(Solution solution, CodeActionPurpose purpose, CodeActionOptionsProvider fallbackOptions, CancellationToken cancellationToken)
+        {
+            foreach (var projectId in solution.ProjectIds)
+            {
+                var project = solution.GetRequiredProject(projectId);
+                if (!CanEnableNullableInProject(project))
+                    continue;
+
+                solution = await EnableNullableReferenceTypesAsync(project, purpose, fallbackOptions, cancellationToken).ConfigureAwait(false);
```

Comment:
nit.  consider doing all projects in parallel and then merging the result at the end.  this seems like a really good case to get multi-core scaling on.